forms-flow-idm/migration/migrate-7.0.0.py

```diff
@@ -0,0 +1,171 @@
+                "description": "Can claim and work on tasks",
+                "composite": False,
+                "clientRole": True,
+                "attributes": {}
+            },
+            {
+                "name": "create_submissions",
+                "description": "Create submissions",
+                "composite": False,
+                "clientRole": True,
+                "attributes": {}
+            },
+            {
+                "name": "create_filters",
+                "description": "Access to create filters",
+                "composite": False,
+                "clientRole": True,
+                "attributes": {}
+            },
+            {
+                "name": "view_tasks",
+                "description": "Access to tasks",
+                "composite": False,
+                "clientRole": True,
+                "attributes": {}
+            },
+            {
+                "name": "manage_dashboard_authorizations",
+                "description": "Manage Dashboard Authorization",
+                "composite": False,
+                "clientRole": True,
+                "attributes": {}
+            },
+            {
+                "name": "view_submissions",
+                "description": "Access to submissions",
+                "composite": False,
+                "clientRole": True,
+                "attributes": {}
+            },
+            {
+                "name": "admin",
+                "description": "Administrator Role",
+                "composite": False,
+                "clientRole": True,
+                "attributes": {}
+            },
+            {
+                "name": "manage_all_filters",
+                "description": "Manage all filters",
+                "composite": False,
+                "clientRole": True,
+                "attributes": {}
+            }
+        ]
+
+        groups_to_role_mappings = {
+            "formsflow/formsflow-admin": [
+                "manage_users",
+                "manage_roles",
+                "manage_dashboard_authorizations",
+                "admin"
+            ],
+            "formsflow/formsflow-client": [
+                "create_submissions", "view_submissions"
+            ],
+            "formsflow/formsflow-designer": [
+                "manage_integrations",
+                "view_designs",
+                "create_designs"
+            ],
+            "formsflow/formsflow-reviewer": [
+                "view_dashboards",
+                "manage_tasks",
+                "view_tasks",
+                "create_filters",
+                "view_submissions",
+                "view_filters"
+            ],
+            "formsflow-analytics": [
+                "view_dashboards"
+            ]
+        }
+
+        # Get client and group IDs
+        client_id = get_client_id(token, "forms-flow-web")
+        client_roles = get_client_roles(token, client_id)
+        update_client_roles(token, client_id, roles_to_update)
```

Comment:
**client_roles** here will have old roles (formsflow-designer,..) and  will not satisfy the if condition  `client_role['name'] == role_to_lookup` on line 161. So roles will be empty. Can you please check.
![image](https://github.com/user-attachments/assets/d262c7a5-1ca6-48ea-b6f9-f5239ef47319)

Reply:
This code basically, matches the client roles which needs to be added to the group and then assign it to the group, so for older roles it doesn't need to get into the if loop as those role assignment exists on the group. Do you see issues with this approach ?

Reply:
Yes. Here `client_roles` will have only old roles(on update_client_roles() - new roles added to the client). So when I executed the script, roles were added to the client, but the new roles were not added to the group(since new roles are not there in client_roles, so not get in the if loop). Executing the script again new roles are added to the group.
![image](https://github.com/user-attachments/assets/f39f3f5f-5076-4b11-a6dd-291ef163c124)